packages/KDropdownMenu/KDropdownItem.vue

```diff
@@ -0,0 +1,165 @@
+<template>
+  <li
+    :class="{
+      'has-divider': type !== 'link' && hasDivider,
+      'disabled': type === 'default' && disabled,
+      'danger': isDangerous,
+      'k-dropdown-selected-option': selected
+    }"
+    class="k-dropdown-item w-100"
+  >
+    <router-link
+      v-if="type === 'link' && to"
+      :data-testid="label"
+      :class="{ 'disabled': disabled, 'has-divider': hasDivider }"
+      :to="!disabled ? to : $router.currentRoute.path"
+      class="k-dropdown-item-trigger"
+    >
+      <slot>{{ label }}</slot>
+    </router-link>
+    <KButton
+      v-else-if="type === 'button'"
+      :disabled="disabled"
+      :class="classList"
+      class="k-dropdown-item-trigger"
+      v-on="listeners"
+    >
+      <slot>{{ label }}</slot>
+    </KButton>
+    <div
+      v-else
+      class="k-dropdown-item-trigger"
+    >
+      <slot>{{ label }}</slot>
+    </div>
+  </li>
+</template>
+
+<script>
+import KButton from '@kongponents/kbutton/KButton.vue'
+
+export default {
+  name: 'KDropdownItem',
+  components: { KButton },
+  props: {
+    item: {
+      type: Object,
+      default: null,
+      // Items must have a label
+      validator: (item) => item.hasOwnProperty('label')
+    },
+    /**
+     * Use this prop to add a divider above the item
+     */
+    hasDivider: {
+      type: Boolean,
+      default: false
+    },
+    isDangerous: {
+      type: Boolean,
+      default: false
+    },
+    disabled: {
+      type: Boolean,
+      default: false
+    },
+    selected: {
+      type: Boolean,
+      default: false
+    },
+    selectionMenuChild: {
+      type: Boolean,
+      default: false
+    }
+  },
+  data () {
+    return {
+      classList: 'btn-link k-button non-visual-button'
+    }
+  },
+  computed: {
+    listeners () {
+      return {
+        ...this.$listeners
+      }
```

Comment:
question: if we're not manipulating the $listeners, do we need to map them to a variable instead of just using `this.$listeners`?

Reply:
I think I was copy/pasting. I'll switch over to `this.$listeners`

Reply:
Actually realized I need this and am now manipulating it